extensions/amp-analytics/0.1/requests.js

```diff
@@ -36,8 +36,16 @@ export class RequestHandler {
    * @param {!../../../src/preconnect.Preconnect} preconnect
    * @param {./transport.Transport} transport
    * @param {boolean} isSandbox
+   * @param {string} requestOrigin
    */
-  constructor(element, request, preconnect, transport, isSandbox) {
+  constructor(
+    element,
+    request,
+    preconnect,
```

Comment:
As you brought up yesterday, we will also need to change when to do the preconnect. 

Before we preconnect after we call `expandTemplate(baseUrl)`, This is based on the assumption that no one is going to dynamically resolve request origin. However since we support `requestOrigin` now, we will need to call preconnect after we call `expandUrlAsync(requestOrigin)` to make sure that we get the correct request origin to preconnect to. Does that make sense?

Reply:
`fire` will now try to preconnect to `requestOrigin` if available instead of `baseUrlTemplate`